mrmustard/lab_dev/circuit_components.py

```diff
@@ -247,6 +288,13 @@ def __matmul__(self, other: CircuitComponent) -> CircuitComponent:
 
         return CircuitComponent._from_attributes("", representation_ret, wires_ret)
 
+    def __lshift__(self, other: CircuitComponent):
+        r"""
+        Projects this component onto another component by computing
+        ``(other >> self.dual).dual``.
+        """
+        return (other >> self.dual).dual
```

Comment:
```suggestion
        return (other.dual >> self.dual).dual
```
No? E.g. if I say `channel.dual << povm`e.g. to evolve the stabilizers backward like Bill does, in MM it would be `(povm.dual >> channel).dual` which makes sense because `povm.dual` is a dm.

Reply:
But I'm thinking about states. If we do `vacuum << vacuum`, in your case we'd have
- `vacuum.dual >> vacuum.dual`, which errors, because I have to objects with input wires only
In my case, it would be `vacuum >> vacuum.dual`, which is what we want

If the lshift for states a different thing than the one you're mentioning above for channels?

Reply:
But `vacuum << vacuum` shouldn't work, just like `vacuum >> vacuum`. You need to flip the in/out wires on one of them

Reply:
Hmm wait. Am I not using the API correctly?